dlext/Sampler.h

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+// This file is part of `hoomd-dlext`, see LICENSE.md
+
+#ifndef SAMPLER_H
+#define SAMPLER_H
+
+#include "hoomd/HalfStepHook.h"
+#include <hoomd/extern/pybind/include/pybind11/pybind11.h>
+#include "dlpack.h"
+
+
+class Sampler : public HalfStepHook
+    {
+    public:
+        //! Constructor
+      Sampler(std::shared_ptr<SystemDefinition> sysdef,
+              pybind11::function python_update);
```

Comment:
I probably wouldn't restrict it to `pybind11::function` and have it as in https://github.com/SSAGESLabs/openmm-dlext/blob/main/openmmapi/include/DLExtForce.h#L34 (just in case the plugin is itself useful for someone as a C++ plugin)

Reply:
Sure, where does the declaration of `Function` come from.
I can't find a reference right now

Reply:
That and some other utils are part of https://github.com/SSAGESLabs/openmm-dlext/blob/main/include/cxx11utils.h, which we should probably have within `dlext/utils.h` (instead of its current contents).

Reply:
So, it is just a version of std::function.
Let me see, what the advantages/disadvantages with pybind11::function are.

Reply:
The only reason to have it as `std::function` instead of `pybind11::function` is that if we should be able to compile without python support in case it's used in another scenario (such as from a pure C++ code base). But, we could change it only when/if such use case happens to appear.

Reply:
I prefer to go with python function for now.
I don't have to specify the signature for that.
Which may or may not be a less beautiful design overall.